### ISSUE TITLE
Make `vue/no-ref-as-operand` also consider `computed`

### ISSUE THREAD

**Issue by DanSnow** [vuejs/eslint-plugin-vue#2114]
**What rule do you want to change?**
`vue/no-ref-as-operand`

**Does this change cause the rule to produce more or fewer warnings?**
more warning

**How will the change be implemented? (New option, new default behavior, etc.)?**
Make `vue/no-ref-as-operand` also consider `computed`

**Please provide some example code that this change will affect:**

```ts
const count = ref(0)
const isEven = computed(() => count.value % 2 === 0)

watchEffect(() => {
  // warning at here
  if (isEven) {
    console.log('is even')
  }
})

```

**What does the rule currently do for this code?**
no warning

**What will the rule do after it's changed?**
Warning about incorrect usage for `computed`

**Additional context**

**Comment by ota-meshi** [comment]
I can't reproduce it. I think `isEven` is already warned.

https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJxNT8uqwjAQ/ZVD4GIKYi8uL8adf5FNiRMNtJOSTOqi9N9N6BXcDec5Z1Uu3kn9KcuX7FKYBZmkzFfLYZpjEqxI5I9wcZqL0P2I1yDuefOenGCDT3GCVUshWzNc5CxVW1hgmlH/dh805NtCXOFPlNYdzHVXn5ZhLIQfnGGMQXNZ/mr6166Wgb6vPyQO/MAgeFKihgYPvVd0uwxovXGk0xgf+hAyqJKHrnGb5a0el37fXNeq7Q3CT1lg

**Comment by DanSnow** [comment]
I tried it again. We didn't see a warning because we use [unplugin-auto-import](https://github.com/antfu/unplugin-auto-import).

I tried on the document's interactive demo

without import => no warning
![image](https://user-images.githubusercontent.com/5575082/227762266-fae0f31f-6f97-4ef1-92a7-cd883c1fb466.png)

with import => has warning
![image](https://user-images.githubusercontent.com/5575082/227762302-e4af50e0-6e99-4294-a1dc-3c11404e7e36.png)

**Comment by ota-meshi** [comment]
Duplicate of #1969. Follow #1969.

**Comment by DanSnow** [comment]
Thanks
